plugins/documents/app/controllers/documents_controller.rb

```diff
@@ -54,10 +55,14 @@ def destroy
 
   def show
     @document = Document.find(params[:id])
-    filename = @document.name
-    unless filename.include? '.'
-      filename += '.' + MIME::Types[@document.mime].first.preferred_extension
+    send_data(@document.data, filename: @document.filename, type: @document.mime)
```

Comment:
Do we need the mime-type check on download as well? Because if the list changes, the old files are still available.
I leave that to your consideration, as long as you thought about it.

Reply:
no, but i think that we should be able to download a file always. even when the list of allowed type changed in the meantime.

Reply:
Ok :+1: